syntax = "proto3";

message ProductRanking {
  int32 id = 1;
  string title_index = 2;
  string weight_index = 3;
  string brand_index = 4;
}

service ProductRankingService {
  rpc List (ListRequest) returns (ListResponse) {}
  rpc CreateItem (CreateItemRequest) returns (CreateItemResponse) {}
}

message ListRequest {
  string title_index = 1;
  string weight_index = 2;
  string brand_index = 3;
}

message ListResponse {
  repeated ProductRanking product_rankings = 1;
}

message CreateItemRequest {
  int32 product_ranking_id = 1;
  int32 circular_product_id = 2;
}

message CreateItemResponse {
  int32 product_ranking_item_id = 1;
}